### ISSUE TITLE
[rllib] Ray workers access to a common file "target_spec.csv"?

### ISSUE THREAD

**Issue by ernest2015** [ray-project/ray#9311]
How can Ray workers access a common file or a common memory buffer?

ray, version 0.9.0.dev0
Python 3.6.9
Ubuntu 18.04

I have a file "target_spec.csv" which contains about 100 target specifications for a Reinforcement problem. I want each worker to read a file, take some specs from it, load them to a memory (numpy array) and then store the remainder of specs back in to the file (choose without replace like this ->  np.random.choice(data, samplesPereWorker, replace=False) and remove from a file).

So the file "target_spec.csv" now contains less specs as before. 

I have a custom gym environment and a "ray" script (ray.init() run("PPO", "env": myEnv, ... "num_workers": 7 ...) that uses this environment. 

File "target_spec.csv" is read from gym environment (in init part of gym class). 
If that file ("target_spec.csv") is also written from gym (savetxt("target_spec.csv", data, delimiter=',')) by a worker there is no problem to read it. But if the worker just want to read it (data = loadtxt("target_spec.csv", delimiter=',') -> the file "target_spec.csv" is already stored in the same folder as the script) then it complains:

    raise IOError("%s not found." % path)
OSError: target_spec.csv not found.

although the file is there.
??

So in short: I'd like to achieve that workers can share some "target specifications" and each of them takes some part of them. I don't want each of workers to create its own file "target_spec.csv". It would be OK if 1st worker create specifications and all others take part from that file or it creates a shared buffer that is reachable from other workers. 

I also found something related to share memory (https://github.com/ray-project/ray/issues/3828) but provided links are broken. 

How can I get the name of a worker in a (ex. in a gym script) when it runs a gym environment?

And finally: I have "num_workers": 7. Does this mean that only 6 workers create a "gym" instance and the 7th is just a coordinator?

Thanks,
Ernest

**Comment by ernest2015** [comment]
Oh, yes, I made a mistake, the root path for the file was changed to ~\ray_results\...
But answers to other questions are welcome.
